rust/transaction-filter/src/traits.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::Error;
+
+/// Simple trait to allow for filtering of items of type T
+pub trait Filterable<T> {
+    /// Whether this filter is correctly configured/initialized
+    /// Any call to `is_valid` is responsible for recursively checking the validity of any nested filters
+    fn is_valid(&self) -> Result<(), Error>;
```

Comment:
I imagine the reason we have `is_valid` is because we can't do it in constructors because these structs might get built directly from configuration, and we can't do it in the parsing logic because serde doesn't provide powerful enough attributes to let us do validity checking at parsing time, yeah?

Reply:
in the constructors of what? 

My assumption for these filters is they will be around for a long time, and different refs will flow through them- there's only ever one copy of these

Injecting it into the serialization would be on the GRPC side, which would be inside of poem, and I wouldn't want to mess with that 😬